transforms/language/ml_filter/kfp_ray/ml_filter_wf.py

```diff
@@ -0,0 +1,236 @@
+        "runtime_worker_options": str(actor_options),
+        "runtime_pipeline_id": runtime_pipeline_id,
+        "runtime_job_id": runtime_job_id,
+        "runtime_code_location": str(runtime_code_location),
+        "ml_filter_column_prefix": ml_filter_column_prefix,
+        "ml_filter_lang_column_name": ml_filter_lang_column_name,
+        "ml_filter_config": ml_filter_config,
+        "ml_filter_ignore_missing_columns": ml_filter_ignore_missing_columns,
+    }
+
+
+# KFPv1 and KFP2 uses different methods to create a component from a function. KFPv1 uses the
+# `create_component_from_func` function, but it is deprecated by KFPv2 and so has a different import path.
+# KFPv2 recommends using the `@dsl.component` decorator, which doesn't exist in KFPv1. Therefore, here we use
+# this if/else statement and explicitly call the decorator.
+if os.getenv("KFPv2", "0") == "1":
+    compute_exec_params_op = dsl.component_decorator.component(
+        func=compute_exec_params_func, base_image=base_kfp_image
+    )
+else:
+    compute_exec_params_op = comp.create_component_from_func(func=compute_exec_params_func, base_image=base_kfp_image)
+
+# create Ray cluster
+create_ray_op = comp.load_component_from_file(component_spec_path + "createRayClusterComponent.yaml")
+# execute job
+execute_ray_jobs_op = comp.load_component_from_file(component_spec_path + "executeRayJobComponent.yaml")
+# clean up Ray
+cleanup_ray_op = comp.load_component_from_file(component_spec_path + "deleteRayClusterComponent.yaml")
+
+# Task name is part of the pipeline name, the ray cluster name and the job name in DMF.
+TASK_NAME: str = "ml_filter"
+
+
+@dsl.pipeline(
+    name=TASK_NAME + "-ray-pipeline",
+    description="filter using a per-language table of conditions",
+)
+def ml_filter(
+    # Ray cluster
+    ray_name: str = "ml_filter-kfp-ray",  # name of Ray cluster
+    ray_run_id_KFPv2: str = "",  # Ray cluster unique ID used only in KFP v2
+    # Add image_pull_secret and image_pull_policy to ray workers if needed
+    ray_head_options: dict = {"cpu": 1, "memory": 4, "image": task_image},
+    ray_worker_options: dict = {
+        "replicas": 2,
+        "max_replicas": 2,
+        "min_replicas": 2,
+        "cpu": 2,
+        "memory": 4,
+        "image": task_image,
+    },
+    server_url: str = "http://kuberay-apiserver-service.kuberay.svc.cluster.local:8888",
+    # data access
+    data_s3_config: str = "{'input_folder': 'test/ml_filter/input/', 'output_folder': 'test/ml_filter/output/'}",
+    data_s3_access_secret: str = "s3-secret",
+    data_max_files: int = -1,
+    data_num_samples: int = -1,
+    data_checkpointing: bool = False,
+    # orchestrator
+    runtime_actor_options: dict = {"num_cpus": 0.8},
+    runtime_pipeline_id: str = "pipeline_id",
+    runtime_code_location: dict = {"github": "github", "commit_hash": "12345", "path": "path"},
+    # ml_filter parameters
+    ml_filter_column_prefix: str = "",
```

Comment:
The current template is used for running CI/CD and will pull the files from the minIO repo that was populated with the test files from test-data/input.
```suggestion
    ml_filter_column_prefix: str = "e_",
```